ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Protocol.hs

```diff
@@ -225,39 +219,36 @@ check :: forall xs. (CanHardFork xs, HasCallStack)
       -> Ticked (ChainDepState (HardForkProtocol xs))
       -> Maybe (HardForkIsLeader xs)
 check HardForkConsensusConfig{..}
```

Comment:
Add postcondition: if the result is `Just`, then `HardForkCanBeLeader` and the ticked chain dep state must be in the same era. The returned `HardForkIsLeader` is from that same era.